# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/inspector/InspectorFontFace.cpp
# Commit: fbfb5b7dfb30
# Full Hash: fbfb5b7dfb30b42d6debd6f2808b0f8e8bb4f868
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-06-29 03:44:30
# Regressor Bug: 1771493
# File Overlap Count: 1
# Description:
#   Bug 1771493 - Part 2. Merge FontFaceSetImpl and gfxUserFontSet into a single class. r=emilio
#   
#   This patch merges FontFaceSetImpl and gfxUserFontSet into the same
#   class. This reduces the level of indirection and in theory makes it
#   easier to manage future threading issues.
# ==============================================================================

diff -r 93b7e171c74a -r fbfb5b7dfb30 layout/inspector/InspectorFontFace.cpp
--- a/layout/inspector/InspectorFontFace.cpp	Tue Jun 28 14:27:20 2022 +0000
+++ b/layout/inspector/InspectorFontFace.cpp	Tue Jun 28 14:27:21 2022 +0000
@@ -69,13 +69,10 @@
     // in the relevant font group's user font set
     RawServoFontFaceRule* rule = nullptr;
     if (mFontEntry->IsUserFont()) {
-      FontFaceSet::UserFontSet* fontSet =
-          static_cast<FontFaceSet::UserFontSet*>(mFontGroup->GetUserFontSet());
-      if (fontSet) {
-        FontFaceSetImpl* fontFaceSet = fontSet->GetFontFaceSet();
-        if (fontFaceSet) {
-          rule = fontFaceSet->FindRuleForEntry(mFontEntry);
-        }
+      auto* fontFaceSet =
+          static_cast<FontFaceSetImpl*>(mFontGroup->GetUserFontSet());
+      if (fontFaceSet) {
+        rule = fontFaceSet->FindRuleForEntry(mFontEntry);
       }
     }
     if (rule) {